pkg/api/message/v1/service.go

```diff
@@ -395,3 +399,11 @@ func (s *Service) BatchQuery(ctx context.Context, req *proto.BatchQueryRequest)
 		Responses: responses,
 	}, nil
 }
+
+func getMaxRows(contentTopics []string) int {
+	if len(contentTopics) == 1 && topic.IsUserPreferences(contentTopics[0]) {
```

Comment:
Maybe put a Todo or comment in here to mention that this is temporary and should be removed. Sometimes I like adding a link to the PR as well for future people/me.

Reply:
Good call